File: LeetCode/0682-baseball-game/0682-baseball-game.cpp
class Solution {
public:
    int calPoints(vector<string>& operations) {
        vector<int> score;
        for (const auto& op : operations) {
            if (op == "C") {
                score.pop_back();
            } else if (op == "D") {
                score.push_back(score.back() * 2);
            } else if (op == "+") {
                int n = score.size();
                score.push_back(score[n-1] + score[n-2]);
            } else {
                score.push_back(stoi(op));
            }
        }
        int sum = 0;
        for (const auto& s : score) {
            sum += s;
        }
        return sum;
    }
};

File: LeetCode/0682-baseball-game/0682-baseball-game.py
class Solution:
    def calPoints(self, operations: List[str]) -> int:
        score = []
        for op in operations:
            if op == "C":
                score.pop()
            elif op == "D":
                score.append(score[-1] * 2)
            elif op == "+":
                score.append(score[-1] + score[-2])
            else:
                score.append(int(op))
        return sum(score)